.github/workflows/fetch_all_tools.yaml

```diff
@@ -81,6 +77,10 @@ jobs:
       - uses: actions/checkout@v4
         with:
           ref: main #pull latest code produced by job 1, not the revision that started the workflow (https://github.com/actions/checkout/issues/439)
+        name: checkout results
+      - run: |
+          git checkout results -- results
+
```

Comment:
```suggestion
      - name: Checkout results
        uses: actions/checkout@v4
        with:
          path: results
          ref: results
```

Reply:
I tried that, but the output would be that the results folder is a subfolder of `results` as `path` specifies a `Relative path under $GITHUB_WORKSPACE to place the repository`. This below works, but is not so elegant I feel. But according to https://github.com/actions/checkout/issues/1430 that seems to be the way atm.
```
    steps:
    - name: Checkout main
      uses: actions/checkout@v4
    - name: Checkout results
      uses: actions/checkout@v4
      with:
        ref: results
        sparse-checkout: results
        path: .tmp
    - name: Move results to root #https://github.com/actions/checkout/issues/1430
      run: |
          mv tmp/results results &&
          rm .tmp
```

Reply:
Why do you want to use a sparse checkout in the first place?

I thought the **result files** would all be on the `results` branch. If you put the **result files** on that branch directly, and not under some sub-directory, then the **result files** should be checked out as `/results`, I'd expect. Am I missing something?

Reply:
nope I was missing something, I was assuming a subfolder, but it is indeed as you describe, so that should work, will adapt the code for it